### ISSUE TITLE
[Feature request] Fullscreen mode

### ISSUE THREAD

**Issue by muranyia** [ahlstromcj/sequencer64#65]
By no means to overwhelm you, rather to throw this into the buffer so you can evaluate and eventually prioritize... :)
I'm thinking about putting sequencer64 on a dedicated PI box with a small LCD display, and fullscreen mode (perhaps switchable by F11 and/or a command line option) would come very handy.
Thank you for all the great work you've done so far, sir!

**Comment by ahlstromcj** [comment]
Hmmmm, it might be nice to make all the windows dockable into the main
window with tabs to switch between them, as a start.

-------- muranyia 15:45 Fri 24 Mar      --------

>By no means to overwhelm you, rather to throw this into the buffer so you can evaluate and eventually prioritize... :)
>I'm thinking about putting sequencer64 on a dedicated PI box with a small LCD display, and fullscreen mode (perhaps switchable by F11 and/or a command line option) would come very handy.
>Thank you for all the great work you've done so far, sir!
>
>-- 
>You are receiving this because you are subscribed to this thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/ahlstromcj/sequencer64/issues/65

-- 
Q:	What's the contour integral around Western Europe?
A:	Zero, because all the Poles are in Eastern Europe!

**Comment by muranyia** [comment]
I usually don't like tabs because they take up space. (My 2 cents)

**Comment by ahlstromcj** [comment]
Can you give me a light storyboard about how you'd like it to work?

For example:

   Either set full-screen in the configuration file, run seq64 with the
   --full-screen option, or hit the F11 key to enter that mode.  What
   appears is the main window.

   Open up a pattern.  Edit.  When done, close the pattern.

   Open up the Song Editor. Edit.  When done, close the Song Editor.

In other words, the windows simply overlay each other, and you work
basically in one window at a time.

Anyway, what ever ideas you have....

-------- muranyia 12:00 Sat 25 Mar      --------

>I usually don't like tabs because they take up space. (My 2 cents)
>
>-- 
>You are receiving this because you commented.
>Reply to this email directly or view it on GitHub:
>https://github.com/ahlstromcj/sequencer64/issues/65#issuecomment-289232200

-- 
Abandon the search for Truth; settle for a good fantasy.

**Comment by muranyia** [comment]
Yeah, either
- a single window goes fullscreen and the other windows stay normal (I think I'd prefer this, and you wouldn't have to design a fullscreen mode for every window type), or
- every window goes fullscreen until fullscreen is revoked.
We'd still have ALT+TAB to rotate windows and eventually some sequencer64-specific shortcuts to bring up specific windows.
As for the main window, I image fullscreen simply stretching (magnifying) the whole content to the entire screen.

**Comment by simonvanderveldt** [comment]
> Hmmmm, it might be nice to make all the windows dockable into the main
> window with tabs to switch between them, as a start.

This would be so nice! Not sure how feasible it would be though. All those separate windows are a bit annoying to work with :)

**Comment by milkmiruku** [comment]
Would it be possible to centre sets?

With a maximised window, I get

![seq64_left](https://user-images.githubusercontent.com/108225/27773992-45f98cd0-5f7f-11e7-8812-8d9dbabc0c18.png)

Fullscreen is

![seq_full](https://user-images.githubusercontent.com/108225/27773997-6a65a900-5f7f-11e7-88f9-7477ab755fd3.png)
